JBPM-8044: [Stunner] Docking is not working when the preview panel is opened.

romartin opened the pull request:
Hey @hasys @tiagodolphine 

See https://github.com/kiegroup/kie-wb-common/pull/2345/

Tests here - https://github.com/kiegroup/lienzo-tests/pull/56

Thanks!

hasys commented on src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresDockingControlImpl.java:
```diff
@@ -157,6 +160,9 @@ private Point2D computeCandidateLocation(WiresShape shape) {
 
     private Point2D getCandidateLocation(WiresShape shape) {
+        if (null == m_intersection) {
+            return null;
```
Good catch, @jomarko,

Yes, I found one possible null pointer for this path:
`WiresShapeControlImpl#execute` -> `WiresDockingControlImpl#getCandidateLocation()` -> `WiresDockingControlImpl#getCandidateLocation(WiresShape)`.

If this path is not reachable so far, it can be in the future. @romartin, is it possible to protect us from it now and in the future?

romartin commented:
Hey @jomarko @hasys  , 

Thanks for the review! 

Just as a note let you know that the `m_intersection` represents the actual intersection point with parent, if any, so this means that not all calls found on the code that could produce an NPE are wrong and should be null-checked, because the intersection is only set when interacting with the canvas, but that's not always the situation, for example, on the preview widhget, which it also uses the shape controls (docking, etc), in order to reproduce the main canvas state, but the user is not really interacting with those controls (with the preview). 

So for example on the first sentence in the `getAdjust` method, which should not really be called when the user is not interacting with the shape, the condition is  `isEnabled() && m_intersection != null`, so it's already checking the `null` value on `m_intersection`. 

Anyway I've fixed both situations you mention and added also tests. Please these bugs are also present on the candidate release yet, so would appreciate some feedback ASAP. Thanks guys!!

romartin commented:
hey @hasys @jomarko please remember to check this, it's urgent as it's targeted for next 7.2.1 release. Thanks!
